### ISSUE TITLE
Merton model - Global solver: Y_0 doesn't update

### ISSUE THREAD

**Issue by janrems** [ZakariaBensaid/DeepFBSDEJSolvers#1]
When runing mainMerton and using Global Solver on the pre-set configuration I get no errors, however the value of Y_0 doesn't update throughout the training even though the loss does decreas. Hence I believe the training of the neural network for prediction of diffusion and jump components of BSDE work but the update of initial value Y_0 doesn't.

**Comment by ZakariaBensaid** [comment]
Thank you for your comment ! I suggest running the codes with the following tensorflow version 2.10.0 and tensorflow_probability version 0.17.0. to avoid this kind of issues. For example, as of 2.11, it is no longer possible to minimize trainable variables with partially known shapes. This explains why the value of Y_0 doesn't update in your setting (we defined it as follows: self.Y0= tf.Variable(tf.keras.initializers.GlorotNormal()([]),  trainable = True, dtype=tf.float32)). For this specific issue, you have to specify the dimension of Y0 in the Net class: 
self.Y0= tf.Variable(tf.keras.initializers.GlorotNormal()(shape=(1,)),  trainable = True, dtype=tf.float32),
and the following updates for the Global solver:
gradients= tape.gradient(objFunc, [self.modelKerasUZ.Y0] + self.modelKerasUZ.trainable_variables + self.modelKerasGam.trainable_variables)
optimizer.apply_gradients(zip(gradients,  [self.modelKerasUZ.Y0] + self.modelKerasUZ.trainable_variables+ self.modelKerasGam.trainable_variables)).
